src/amo/components/AddonReviewListItem/index.js

```diff
@@ -0,0 +1,121 @@
+/* @flow */
+/* eslint-disable react/sort-comp */
+import React from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import AddonReview from 'amo/components/AddonReview';
+import translate from 'core/i18n/translate';
+import { isAuthenticated } from 'core/reducers/user';
+import { nl2br, sanitizeHTML } from 'core/utils';
+import Button from 'ui/components/Button';
+import LoadingText from 'ui/components/LoadingText';
+import Rating from 'ui/components/Rating';
+import type { UserReviewType } from 'amo/actions/reviews';
+import type { UserStateType } from 'core/reducers/user';
+
+import './styles.scss';
+
+type PropsType = {|
+  isAuthenticated: boolean,
+  i18n: Object,
+  review: UserReviewType,
+  siteUser: UserStateType,
+|};
+
+export class AddonReviewListItemBase extends React.Component {
+  props: PropsType;
+  state: {| editingReview: boolean |};
+
+  constructor(props: PropsType) {
+    super(props);
+    this.state = { editingReview: false };
+  }
+
+  onClickToEditReview = (event: SyntheticEvent) => {
+    event.preventDefault();
+    this.setState({ editingReview: true });
+  }
+
+  onEscapeReviewOverlay = () => {
+    // Even though an escaped overlay will be hidden, we still have to
+    // synchronize our show/hide state otherwise we won't be able to
+    // show the overlay after it has been escaped.
```

Comment:
This is a bit lame since we have a chain of callbacks now. I explained the issue in https://github.com/mozilla/addons-frontend/issues/3222 but I think using callbacks is a quicker fix for now.